core/src/main/java/com/linecorp/armeria/client/RedirectingClient.java

```diff
@@ -573,24 +567,87 @@ CompletableFuture<HttpResponse> responseFuture() {
             return responseFuture;
         }
 
+        boolean addRedirectSignature(RequestTarget nextReqTarget, HttpMethod nextMethod) {
+            if (redirectSignatures == null) {
+                redirectSignatures = new LinkedHashSet<>();
+            }
+
+            if (!isAddedOriginalSignature) {
+                final String originalProtocol = ctx.sessionProtocol().isTls() ? "https" : "http";
+                final RedirectSignature originalSignature = new RedirectSignature(originalProtocol,
+                                                                                  ctx.authority(),
+                                                                                  request.headers().path(),
+                                                                                  request.method());
+                redirectSignatures.add(originalSignature);
+                isAddedOriginalSignature = true;
+            }
```

Comment:
Question) Could we add `originalSignature` as `redirectSignatures` is allocated?
For example:
```suggestion
            if (redirectSignatures == null) {
                redirectSignatures = new LinkedHashSet<>();

                final String originalProtocol = ctx.sessionProtocol().isTls() ? "https" : "http";
                final RedirectSignature originalSignature = new RedirectSignature(originalProtocol,
                                                                                  ctx.authority(),
                                                                                  request.headers().path(),
                                                                                  request.method());
                redirectSignatures.add(originalSignature);
            }
```

Reply:
yes, I will also remove `isAddedOriginalSignature` flag.